### PR TITLE
Feat/team section

### PR DESCRIPTION
**Pull request by rolandiartmeladze**
# Pull Request Template for ArtOa

## Description

Please provide a summary of the changes in this pull request:

- **What does this PR do?**  
  This PR updates the `team section` component to improve its design and functionality. It includes enhancements for responsiveness, styling updates, and optimized rendering logic.

- **Why are these changes necessary?**  
  The updates address feedback for a more visually appealing layout and seamless user experience. Additionally, the refactoring improves maintainability and reduces redundant code.

- **Relevant links or references:**  
  N/A

Fixes: [#56]

---

## Type of Change

Please select the type of change this PR introduces (tick the appropriate box):

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)  
- [x] ✨ New feature (non-breaking change that adds functionality)  
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality not to work as expected)  
- [ ] 🔄 Refactor (code or file structure changes without affecting functionality)  
- [ ] 📝 Documentation update (changes or additions to documentation)  
- [ ] 🚀 Performance improvement (optimization in code or resources)  

---

## Checklist

Please ensure the following tasks have been completed before requesting a review:

- [x] My code follows the style guidelines of this project.  
- [x] I have run tests to ensure my changes work as expected.  
- [x] I have added/updated necessary tests or documentation (if applicable).  
- [x] I have self-reviewed my code and refactored any areas that need improvement.  
- [x] There are no console errors or warnings related to my changes.  
- [x] The PR includes a clear, detailed title and description.  
- [x] Any relevant issues have been linked or mentioned in this PR.  
- [x] I have reviewed the acceptance criteria and met all requirements.  
- [x] I have considered edge cases and errors that may occur due to my changes.  

---

## Testing Instructions

Please describe how your changes can be tested:

1. **Test Setup:**  
   

2. **Test Steps:**  
  

---

## Screenshots / Visual Changes

Please attach any relevant screenshots, videos, or GIFs to illustrate the impact of this change (if applicable):  

**Example Screenshot:**  

 

---

## Additional Notes

Is there anything else the reviewer should know about this PR? Any known issues, challenges, or decisions made during implementation?  
.

### CODE REVIEW

File: src/config/team.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import teamImage1 from "@/../public/team1.svg";
+import teamImage2 from "@/../public/team2.svg";
+import teamImage3 from "@/../public/team3.svg";
+
+export const TeamConfig = [
```
Comment (mshubitidze) [inline src/config/team.ts:5]: this variable should be lowercase

File: src/config/team.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    name: "Name",
+    surname: "Surname",
+    role: "Founder & CEO",
+    image: teamImage1,
+  },
+  {
+    name: "Name",
+    surname: "Surname",
+    role: "Co-Founder & CEO",
+    image: teamImage2,
```
Comment (mshubitidze) [inline src/config/team.ts:16]: instead of image, this can be `href` which directly points to the public folder `/team/image1.svg|png`

File: src/types/employee.ts
Comment (mshubitidze) [inline src/types/employee.ts:1]: creating a separate type for an employee card doesnt really make sense here, we can just infer it from the teamConfig object if necessary, eg.: type Employee = TeamConfig[number]

File: src/config/team.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import teamImage1 from "@/../public/team1.svg";
```
Comment (mshubitidze) [inline src/config/team.ts:1]: this is incorrect import syntax, not necessary either

File: src/components/homepage/artoa-team-section.tsx [hunk 4]
```diff
@@ -1,3 +1,21 @@
+import { TeamConfig } from "@/config/team";
+import { TeamItemCard } from "../team/team-card";
+
 export function ArtoaTeamSection() {
-  return <section className="container mx-auto">artoa-team-section</section>;
+  const teamMembers = Object.values(TeamConfig);
```
Comment (mshubitidze) [inline src/components/homepage/artoa-team-section.tsx:5]: why are we using Object.values here?
you can just map over the teamConfig array directly

File: src/components/ui/team.tsx
Comment (mshubitidze) [inline src/components/ui/team.tsx:1]: good job figuring out forwardRef!
except in case of team member cards, it might not be necessary to create a reusable component since it will not be used anywhere but the TeamSection

File: src/components/ui/team.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    {...props}
+  />
+));
+TeamItemImage.displayName = "TeamItemImage";
+
+export const TeamContent = React.forwardRef<
+  HTMLDivElement,
+  React.HTMLAttributes<HTMLDivElement>
+>(({ className, ...props }, ref) => (
+  <div ref={ref} className={cn("p-4 pt-0", className)} {...props} />
```
Comment (mshubitidze) [inline src/components/ui/team.tsx:43]: why are you setting default padding here, just to remove it when using this component in team-card.tsx?

**Review by mshubitidze — CHANGES_REQUESTED** [review]
overall, good work, can use some cleanup tho

**Review by chubinaaa — APPROVED** [review]
